### PR TITLE
Create python script to rename wikis with one command (T11621) 

### PR DESCRIPTION
**Pull request by RhinosF1**

### CODE REVIEW

File: modules/mediawiki/files/cookbooks/rename-wiki.py [hunk 1]
```diff
@@ -11,42 +22,54 @@
...
+    sys.exit(1)  # noqa: R503
 
-def main():
-    parser = argparse.ArgumentParser(description='Executes the commands needed to rename wikis')
-    parser.add_argument('--oldwiki', required=True, help='Old wiki database name')
-    parser.add_argument('--newwiki', required=True, help='New wiki database name')
 
-    args = parser.parse_args()
+def get_db_cluster(oldwiki_db: str) -> str:
+    command = generate_salt_command('db131*', f'cmd.run "mysql -e \'SELECT wiki_dbcluster FROM mhglobal.cw_wikis WHERE wiki_dbname = "{oldwiki_db}" \' "')
```
Comment (Reception123) [inline modules/mediawiki/files/cookbooks/rename-wiki.py:39]: I think this should be:
command = generate_salt_command('db131*', f'cmd.run \'mysql -e "SELECT wiki_dbcluster FROM mhglobal.cw_wikis WHERE wiki_dbname = \'{oldwiki_db}\'" \'')

**Comment by Reception123** [comment]
I've tried various things but it seems python/salt don't like escaping. We could potentially replace the SQL with a RemoteWiki getDBcluster but that would need a script.

File: modules/mediawiki/files/cookbooks/rename-wiki.py [hunk 2]
```diff
@@ -11,42 +22,54 @@
...
-    execute_salt_command(oldwiki_cluster, f'mysqldump {oldwiki_db} > oldwikidb.sql')
-    execute_salt_command(oldwiki_cluster, f"mysql -e 'CREATE DATABASE {newwiki_db}'")
-    execute_salt_command(oldwiki_cluster, f"mysql -e 'USE {newwiki_db}; SOURCE /home/$user/oldwikidb.sql'")
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f'mysqldump {oldwiki_db} > oldwikidb.sql'))
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"mysql -e 'CREATE DATABASE {newwiki_db}'"))
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"mysql -e 'USE {newwiki_db}; SOURCE /home/$user/oldwikidb.sql'"))
 
     # Step 3: Execute MediaWiki rename script
-    execute_salt_command('mwtask141', f'sudo -u www-data php /srv/mediawiki/w/extensions/CreateWiki/maintenance/renameWiki.php --wiki=loginwiki --rename {oldwiki_db} {newwiki_db} $user')
+    execute_salt_command(salt_command=generate_salt_command('mwtask141', f'sudo -u www-data php /srv/mediawiki/w/extensions/CreateWiki/maintenance/renameWiki.php --wiki=loginwiki --rename {oldwiki_db} {newwiki_db} $user'))
```
Comment (Universal-Omega) [inline modules/mediawiki/files/cookbooks/rename-wiki.py:63]: Nerds to support multiversion and use mwtask181

File: modules/mediawiki/files/cookbooks/rename-wiki.py [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+    c4: str
+    c5: str
+
 
 # Define the mapping of db clusters to db names
-db_clusters = {
+db_clusters: DbClusterMap = {
     'c1': 'db131',
     'c2': 'db101',
     'c3': 'db142',
```
Comment (Universal-Omega) [inline modules/mediawiki/files/cookbooks/rename-wiki.py:19]: update clusters

File: modules/mediawiki/files/cookbooks/rename-wiki.py (hunk 1 above)
Comment (Universal-Omega) [inline modules/mediawiki/files/cookbooks/rename-wiki.py:39]: Update server

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

**Review by Universal-Omega — APPROVED** [review]

**Pull request merged**